fvm/evm/offchain/sync/replay.go

```diff
@@ -165,9 +165,10 @@ func ValidateResult(
 		return fmt.Errorf("gas consumption mismatch %d != %d", res.GasConsumed, txEvent.GasConsumed)
 	}
 
-	// check error msg
-	if errMsg := res.ErrorMessageWithRevertReason(); errMsg != txEvent.ErrorMessage {
```

Comment:
Why not comparing `res.ErrorMsg() != txEvent.ErrorMessage`? We could still include `res.ErrorMessageWithRevertReason()` in the error message if doesn't match.

Or could we add both checks (errorCode + ErrorMsg)?

Reply:
> Why not comparing `res.ErrorMsg() != txEvent.ErrorMessage`? We could still include `res.ErrorMessageWithRevertReason()` in the error message if doesn't match.
> 
This approach was actually error prone from the beginning. Most of the error objects are actually defined in the official `github.com/ethereum/go-ethereum` codebase, so if they change any of the error messages, we could end up with the same situation, when we'll have to update the `go-ethereum` dependency.
By using the `errorCode` that validation is not sensitive to the error message string content, which can change for whatever reason.

> I think it was broken because we added revert reason into the error message. no?
> 
 That is correct, it was broken by that.

Reply:
I agree using the error message will become unmaintainable. are the error codes granular enough?